src/main/java/org/mule/extension/file/internal/FileOperations.java

```diff
@@ -156,12 +160,16 @@
   public void write(@Config FileConnectorConfig config, @Connection FileSystem fileSystem,
                     @Path(type = FILE, location = EXTERNAL) String path,
                     @Content @Summary("Content to be written into the file") InputStream content,
-                    @Optional @Summary("Encoding when trying to write a String file. If not set, defaults to the configuration one or the Mule default") @Placement(
-                        tab = ADVANCED_TAB) String encoding,
+                    @Optional @Summary("This parameter is deprecated and will not be taken into account") @Placement(
+                        tab = ADVANCED_TAB) @DisplayName("Encoding (DEPRECATED)") String encoding,
                     @Optional(defaultValue = "true") boolean createParentDirectories,
                     @Optional(defaultValue = "false") @Placement(tab = ADVANCED_TAB) boolean lock, @Optional(
                         defaultValue = "OVERWRITE") @Summary("How the file is going to be written") @DisplayName("Write Mode") FileWriteMode mode) {
-    super.doWrite(config, fileSystem, path, content, encoding, createParentDirectories, lock, mode);
+    if (encoding != null) {
+      LOGGER
+          .warn("File write operation has the 'encoding' parameter set. This parameter is deprecated and will not change the behaviour of the operation");
```

Comment:
`"Deprecated parameter 'encoding' was configured for operation 'file:write'. This parameter will be ignored, not altering the operation behavior"` ?

Reply:
👍